.github/workflows/deploy-and-test.yaml

```diff
@@ -0,0 +1,109 @@
+name: Deploy and Test
+
+on:
+  workflow_call:
+    inputs:
+      docker_image_tag:
+        required: true
+        type: string
+        description: "The Docker image tag to deploy"
+      environment:
+        required: true
+        type: string
+        description: "The environment to deploy to (Development or Staging)"
+      source_repo:
+        required: true
+        type: string
+        description: "Source repository for the image"
+      target_repo:
+        required: true
+        type: string
+        description: "Target repository for the image"
+      rpc_version:
+        required: false
+        type: string
+        default: "v0_7"
+    secrets:
+      ARTIFACTORY_NUBIA_USERNAME:
+        required: true
+      ARTIFACTORY_NUBIA_TOKEN_DEVELOPER:
+        required: true
+      RPC_URL:
+        required: true
+      TEST_ACCOUNT_ADDRESS:
+        required: true
+      TEST_ACCOUNT_PRIVATE_KEY:
+        required: true
+
+concurrency:
+    group: shared_${{ inputs.environment }}_environment
+    cancel-in-progress: false 
+    
+permissions:
+  contents: read  
+
+env:
+  DOCKER_REGISTRY: nethermind.jfrog.io
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ inputs.environment }}
```

Comment:
I'm wondering if we can move some secrets to this environment, but this can be in a different PR if possible.